### ISSUE TITLE
[3.3.5] Ancient gate of the keepers permanently locked

### ISSUE THREAD

**Issue by RaaghGit** [TrinityCore/TrinityCore#17072]
The Ancient gate of the keepers in the Ulduar raid isn't unlocked when the 4 keepers are defeated.
I tested it a few time, but the gate stays locked.

It it not clickable even in GM 

It should unlock when you defeat Freya, Thorim, Mimiron and Odir

GO-Id: 194255 

**Steps to reproduce the problem**: Kill Freya, Thorim, Mimiron and Odir and go to the Hall of Memories where you'll find the gate.

**Branch** 3.3.5

**TDB**: TDB 335.6

**OS**: Windows 7

**TC Revison 9064968a4dee 2016-05-02**

 Joschiwald's scripts were used for Thorim
#15008

**Comment by TrinityCoreBot** [comment]
This issue is not considered valid because of the following reasons:
- No proper commit hash (revision)

Please read http://bit.ly/tc-issuetracker-and-you . Thanks.

**Comment by DDuarte** [comment]
Don't ignore the template that is given to you when creating a new issue...

**Comment by tkrokli** [comment]
Have a look at issue  #17067 (Client crashes on launch - Mac) for a good example of how to use the template provided by TrinityCore. With all the missing information that template can be used to fill in, it is much harder for anyone to follow up your issue and find a solution to it.

**Comment by RaaghGit** [comment]
Thank you, I thought I had to add manually the bookmarks afterwards.
I think I added all the necessary information now.

**Comment by tkrokli** [comment]
Your description is still a bit lacking. You could point out if this is about Ulduar, if you have tested it in 10 Normal, 10 Heroic, 25 Normal or 25 Heroic (include all tested modes) just so any new reader of your issue does not have to be in doubt about where this is. TC core revision hash is still missing.

**Comment by Keader** [comment]
Bug exists, need script from unlock the gate after 4 keepers "die".

**Comment by Aokromes** [comment]
It's obvious since thorim is unscripted.

**Comment by RaaghGit** [comment]
I merged joschiwald's scripts for Thorim and it all works. Thorim also appears to give you his buff for Yogg Saron's encounter.

The gate is not unlocked anyway.

**Comment by Keader** [comment]
yeap, gates still need script...
Something need check if Thorim,Freya,Hodir and Mimiron is done for "unlock" the gate.

**Comment by Kittnz** [comment]
Can you add the commit hash please?

**Comment by RaaghGit** [comment]
I did :)

**Comment by L30m4nc3r** [comment]
Confirm, use @joschiwald Thorim script - door not open.

**Comment by sirikfoll** [comment]
You can try this, requires Thorim PR
```diff
diff --git a/src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.h b/src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.h
index 72dbde8..3aa9d8f 100644
--- a/src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.h
+++ b/src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.h
@@ -301,6 +301,8 @@ enum UlduarGameObjects
     GO_DOODAD_UL_ULDUAR_TRAPDOOR_03         = 194253,
     GO_GIFT_OF_THE_OBSERVER_10              = 194821,
     GO_GIFT_OF_THE_OBSERVER_25              = 194822,
+
+    GO_GATE_OF_THE_KEEPERS                  = 194255,
 };
 
 enum EventIds

diff --git a/src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp b/src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp
index f81d47c..d0d14b6 100644
--- a/src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp
+++ b/src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp
@@ -163,6 +163,7 @@ class instance_ulduar : public InstanceMapScript
             ObjectGuid AlgalonUniverseGUID;
             ObjectGuid AlgalonTrapdoorGUID;
             ObjectGuid GiftOfTheObserverGUID;
+            ObjectGuid KeepersDoorGUID;
 
             // Miscellaneous
             uint32 TeamInInstance;
@@ -228,6 +229,8 @@ class instance_ulduar : public InstanceMapScript
                     instance->SummonCreature(NPC_THORIM_YS, YSKeepersPos[2]);
                 if (_summonYSKeeper[3])
                     instance->SummonCreature(NPC_MIMIRON_YS, YSKeepersPos[3]);
+
+                CheckGateOfTheKeepers();
             }
 
             void OnCreatureCreate(Creature* creature) override
@@ -634,6 +637,9 @@ class instance_ulduar : public InstanceMapScript
                     case GO_GIFT_OF_THE_OBSERVER_25:
                         GiftOfTheObserverGUID = gameObject->GetGUID();
                         break;
+                    case GO_GATE_OF_THE_KEEPERS:
+                        KeepersDoorGUID = gameObject->GetGUID();
+                        break;
                     default:
                         break;
                 }
@@ -772,11 +778,19 @@ class instance_ulduar : public InstanceMapScript
                         break;
                     case BOSS_MIMIRON:
                         if (state == DONE)
+                        {
+                            CheckGateOfTheKeepers();
+
                             instance->SummonCreature(NPC_MIMIRON_OBSERVATION_RING, ObservationRingKeepersPos[3]);
+                        }
                         break;
                     case BOSS_FREYA:
                         if (state == DONE)
+                        {
+                            CheckGateOfTheKeepers();
+
                             instance->SummonCreature(NPC_FREYA_OBSERVATION_RING, ObservationRingKeepersPos[0]);
+                        }
                         break;
                     case BOSS_IRONBRANCH:
                     case BOSS_STONEBARK:
@@ -805,6 +819,8 @@ class instance_ulduar : public InstanceMapScript
                             if (GameObject* HodirChest = instance->GetGameObject(HodirChestGUID))
                                 HodirChest->SetRespawnTime(HodirChest->GetRespawnDelay());
 
+                            CheckGateOfTheKeepers();
+
                             instance->SummonCreature(NPC_HODIR_OBSERVATION_RING, ObservationRingKeepersPos[1]);
                         }
                         break;
@@ -821,7 +837,7 @@ class instance_ulduar : public InstanceMapScript
                                 }
                             }
 
-                            //CheckGateOfTheKeepers();
+                            CheckGateOfTheKeepers();
 
                             instance->SummonCreature(NPC_THORIM_OBSERVATION_RING, ObservationRingKeepersPos[2]);
                         }
@@ -1297,6 +1313,16 @@ class instance_ulduar : public InstanceMapScript
                     InstanceScript::AddDoor(door, add);
             }
 
+            void CheckGateOfTheKeepers()
+            {
+                for (uint8 bossId = BOSS_HODIR; bossId <= BOSS_MIMIRON; ++bossId)
+                    if (GetBossState(bossId) != DONE)
+                        return;
+
+                if (GameObject* door = instance->GetGameObject(KeepersDoorGUID))
+                    door->RemoveFlag(GAMEOBJECT_FLAGS, GO_FLAG_LOCKED);
+            }
+
         protected:
             // Thorim
             ObjectGuid ThorimGUID;

**Comment by joschiwald** [comment]
will be implemented once thorim pr is merged

**Comment by ariel-** [comment]
Should I add this code to the Thorim PR branch? Or better yet, using doordata :P

**Comment by L30m4nc3r** [comment]
@sirikfoll for me not work :o

**Comment by Keader** [comment]
@L30m4nc3r you add thorim PR?
Me and sirikfoll is using it for 3-4 months... and works fine.

**Comment by hlainin** [comment]
The fix doesn't work. Maybe there is another solution?

**Comment by Rushor** [comment]
This https://github.com/TrinityCore/TrinityCore/pull/15008 wasn't merged yet, so the check in CheckGateOfTheKeepers() always fails ofc.

**Comment by Keader** [comment]
Yes, and Ariel already fixed in Thorim PR btw ^^
Waiting Joschiwald merge it.
